File: src/main/java/com/ecoroute/backend/domain/model/Route.java
package com.ecoroute.backend.domain.model;

import java.time.LocalDate;
import java.time.OffsetDateTime;

public record Route(
    Long id,
    Long driverId,
    Long vehicleId,
    LocalDate routeDate,
    RouteStatus status,
    OffsetDateTime estimatedStartTime,
    OffsetDateTime actualStartTime,
    OffsetDateTime actualEndTime,
    Double totalDistanceKm,
    OffsetDateTime createdAt
) {}

File: mobile-app/lib/features/routes/domain/entities/route.dart
class RouteEntity {
  final int id;
  final int driverId;
  final int vehicleId;
  final String routeDate;
  final String status;

  RouteEntity({
    required this.id,
    required this.driverId,
    required this.vehicleId,
    required this.routeDate,
    required this.status,
  });
}
